src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -211,9 +216,40 @@ def evaluate(
                     invalid_indices = [i for i, x in enumerate(is_valid.values) if not x]
                     energy_usage[invalid_indices] = list(np.nan_to_num(energy_usage_for_period[invalid_indices].values))
                     power[invalid_indices] = list(np.nan_to_num(power_for_period[invalid_indices].values))
-                    operational_settings_used[invalid_indices] = [
-                        operational_setting_index + 1 for _ in invalid_indices
-                    ]
+                    operational_settings_used[invalid_indices] = [operational_setting_index for _ in invalid_indices]
+
+        sub_component_results = []
+        for period, operational_settings in temporal_operational_settings.items():
+            variables_map_for_period = variables_map.get_subset_from_period(period)
+            composite_operational_settings = deepcopy(operational_settings[0])
+            for time_index, setting_number in enumerate(operational_settings_used.values):
+                for key, value in operational_settings[setting_number].__dict__.items():
+                    if isinstance(value, list) and len(value) > 0:
+                        if isinstance(value[0], TimeSeries):
+                            for consumer in range(len(value)):
+                                composite_operational_settings.__getattribute__(key)[consumer][time_index] = value[
+                                    consumer
+                                ][time_index].values[0]
+
+            (
+                adjusted_operational_settings,
+                sorted_consumers,
+            ) = ConsumerSystem._get_operational_settings_adjusted_for_crossover(
+                operational_setting=composite_operational_settings,
+                consumers=self._consumers,
+                variables_map=variables_map_for_period,
+            )
+
+            sub_component_results.extend(
+                [
+                    consumer.evaluate(adjusted_operational_setting)
+                    for consumer, adjusted_operational_setting in zip(sorted_consumers, adjusted_operational_settings)
+                ]
+            )
+
+        # Change to 1-index operational_settings_used
+        for idx, value in enumerate(operational_settings_used.values):
```

Comment:
idx?

Reply:
Renamed to index. I think idx is a common short for index.